ee/libcglue/src/fdman.c

```diff
@@ -0,0 +1,161 @@
+/*
+# _____     ___ ____     ___ ____
+#  ____|   |    ____|   |        | |____|
+# |     ___|   |____ ___|    ____| |    \    PS2DEV Open Source Project.
+#-----------------------------------------------------------------------
+# (C)2001, Gustavo Scotti (gustavo@scotti.com)
+# (c) 2003 Marcus R. Brown <mrbrown@0xd6.org>
+# (c) 2023 Francisco Javier Trujillo Mata <fjtrujy@gmail.com>
+# Licenced under Academic Free License version 2.0
+# Review ps2sdk README & LICENSE files for further details.
+*/
+
+/**
+ * @file
+ * fdman.c - Manager for fd.
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include <kernel.h>
+#include "fdman.h"
+
+#ifdef F___fdman_sema
+int __fdman_sema = -1;
+#else
+extern int __fdman_sema;
+#endif
+
+#ifdef F___descriptor_data_pool
+__descriptormap_type  __descriptor_data_pool[__FILENO_MAX];
+#else
+extern __descriptormap_type  __descriptor_data_pool[__FILENO_MAX];
+#endif
+
+#ifdef F___descriptormap
+__descriptormap_type *__descriptormap[__FILENO_MAX];
+#else
+extern __descriptormap_type *__descriptormap[__FILENO_MAX];
+#endif
+
+
+#ifdef F___fdman_init
+void __fdman_init()
+{
+	ee_sema_t sema;
+
+	/* Initialize mutex */
+	sema.init_count      = 1;
+    sema.max_count       = 1;
+    sema.option          = 0;
+	__fdman_sema = CreateSema(&sema);
+	if (__fdman_sema < 0) {
+		abort();
```

Comment:
not really sure what to do here....

Reply:
This is an unusual case, so leaving this as is should be fine.

Semaphore count should be low near the start of execution so shouldn't run out of semaphores here

Reply:
Súper!